addons/account/models/company.py

```diff
@@ -395,8 +402,14 @@ def write(self, values):
             if 'currency_id' in values and values['currency_id'] != company.currency_id.id:
                 if company.root_id._existing_accounting():
                     raise UserError(_('You cannot change the currency of the company since some journal items already exist'))
+            if values.get('country_id') and not company.country_id and self.env.registry.ready or not self.env.registry._init:
+                companies_needs_l10n.append(company)
+
+        res = super(ResCompany, self).write(values)
+        for company in companies_needs_l10n:
+            company.install_l10n_modules()
```

Comment:
@william-andre as per discussion with @tsb-odoo it should be done when `account` is installed

Though I have question here the install localization label only appeared when `account` has installed (may be you can explain what was the functional purpose of the task), still need to check it with the `hr`

Reply:
> it should be done when `account` is installed

I'm talking about the installation of the module itself, only the instanciation of the CoA on the company

Reply:
I will move to the base
@tsb-odoo Are we taking in consideration when a database without demo is created and you don't select the country, and install`account` module it sets the default country is `United State` then we should install `l10n_us` or not? Because we do it this way then it might trigger installation of `l10n_us`

Reply:
@hamo-odoo In this case, I prefer we do not enable any package in the back of the user. Thanks for asking. Nice catch
cc : @william-andre

Reply:
To avoid that, you can add `and self.name != 'account'` in `<ir.module>.write` for the condition to set `_auto_install_template` (in `account`)